### PR TITLE
chore: Upgrade Stellar SDK to 22.0.0-rc.3.2

### PR DESCRIPTION
**Pull request by kovipu**
* Upgrade Stellar SDK to the latest 22.0 release candidate.
* Update Actions to not run on main, as that's unnecessary.
* Fix typo in the scripts.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -34,7 +35,7 @@ jobs:
         uses: Swatinem/rust-cache@v2
 
       - name: Install Stellar
-        run: cargo install --locked stellar-cli --features opt
+        run: cargo install --locked stellar-cli@22.0.0-rc.1 --features opt
```
Comment (Teolhyn) [inline .github/workflows/deploy.yml:38]: Could we get this from Cargo.toml? Then we wouldn't need to remember to change this in future updates.

File: Cargo.toml [hunk 2]
```diff
@@ -6,8 +6,8 @@ resolver = "2"
 members = ["contracts/*"]
 
 [workspace.dependencies]
-soroban-sdk = "21.7.6"
-soroban-token-sdk = "21.7.6"
+soroban-sdk = "22.0.0-rc.3.2"
```
Comment (Teolhyn) [inline Cargo.toml:9]: Now we have rc.1 in actions and 3.2 here. Is it ok?

**Review by Teolhyn — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (kovipu) [reply Cargo.toml:9]: That was the latest version for stellar CLI as that's a different tool from the SDK

Comment (kovipu) [reply Cargo.toml:9]: https://github.com/stellar/stellar-cli/releases

**Pull request merged**